spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/stream/SkipperStreamDeployer.java

```diff
@@ -463,14 +463,18 @@ private List<Template> createGenericTemplate() {
     }
 
     public void undeployStream(String streamName) {
-        Collection<PackageMetadata> packageMetadataResources = this.skipperClient.search(streamName, false);
-        if (!packageMetadataResources.isEmpty()) {
+        Collection<PackageMetadata> packages = this.skipperClient.search(streamName, false);
+        boolean packageExists = packages.stream().map(PackageMetadata::getName).anyMatch(s -> s.equalsIgnoreCase(streamName));
+        if (packageExists) {
             try {
                 this.skipperClient.delete(streamName, true);
             } catch (ReleaseNotFoundException e) {
                 logger.info("Release not found for {}. Deleting the package {}", streamName, streamName);
                 this.skipperClient.packageDelete(streamName);
-            }
+            };
+        } else {
+            // Stream may have package data is corrupted/missing so we attempt to delete all but the package info
```

Comment:
Nitpick: package data `that` is

Reply:
Good catch - actually the comment is outdate altogether - yanked.